sdk/sdk/OBAApplication.h

```diff
@@ -0,0 +1,44 @@
+//
+//  sdk.h
+//  sdk
+//
+//  Created by Dima Belov on 4/25/15.
+//  Copyright (c) 2015 One Bus Away. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "OBAReferencesV2.h"
+#import "OBAModelDAO.h"
+#import "OBAModelService.h"
+#import "OBALocationManager.h"
+
+@interface OBAApplication : NSObject
+
+@property(nonatomic, strong, readonly) OBAReferencesV2 * references;
+@property(nonatomic, strong, readonly) OBAModelDAO * modelDao;
+@property(nonatomic, strong, readonly) OBAModelService * modelService;
+@property(nonatomic, strong, readonly) OBALocationManager * locationManager;
+
+/**
+ *  This block, if set, is called whenever refreshSettings also refreshes a region.
+ */
+@property (nonatomic, copy) dispatch_block_t regionRefreshed;
```

Comment:
I disagree.  This blocks only purpose is to maintain legacy business logic compatibility.   I'd like to see it gone altogether but did not feel a refactor was justified for this PR.  The reason it is still here is because of how coupled OBARegionHelper is with non-extension compatible APIs and could not be extracted easily into another layer.

Sure adding delegation / event system support will make it easier to expand in future, but that is really not the point here.   A delegate can be overwritten just as easily (I can add this block as a param to the factory method if you wish)

Additionally it will add another layer of indirection to the codebase, that is simply not needed to achieve this.

imho, there is no point in trying to be fancy here, this block achieves the requirements with minimal impact.

As far as sync or async, the consumer of this block is abstracted from that detail.  For all we know this class is all async.